src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/publicApi/SubjectAccessRequestServiceTest.kt

```diff
@@ -0,0 +1,225 @@
+
+  private val service = SubjectAccessRequestService(
+    licenceService,
+    licenceRepository,
+    licenceEventRepository,
+    auditEventRepository,
+  )
+
+  @BeforeEach
+  fun reset() {
+    reset(
+      licenceService,
+      licenceRepository,
+      licenceEventRepository,
+      auditEventRepository,
+    )
+  }
+
+  @Test
+  fun `service returns a list of licence summaries by crn`() {
+    whenever(licenceRepository.findAllByNomsId("A12345")).thenReturn(
+      listOf(
+        aLicenceEntity,
+      ),
+    )
+    whenever(licenceService.getLicenceById(1L)).thenReturn(modelLicence)
+
+    val sarContent = service.getSarRecordsById("A12345")
+
+    assertThat(sarContent).isExactlyInstanceOf(SarContent::class.java)
+
+    assertThat(sarContent?.content?.licences?.first()).isEqualTo(modelLicence)
+
+    verify(licenceRepository, times(1)).findAllByNomsId("A12345")
+    verify(licenceService, times(1)).getLicenceById(1L)
+  }
+
+  private companion object {
+
+    val someStandardConditions = listOf(
+      uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.StandardCondition(
+        id = 1,
+        code = "goodBehaviour",
+        sequence = 1,
+        text = "Be of good behaviour",
+      ),
+      uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.StandardCondition(
+        id = 2,
+        code = "notBreakLaw",
+        sequence = 1,
+        text = "Do not break any law",
+      ),
+      uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.StandardCondition(
+        id = 3,
+        code = "attendMeetings",
+        sequence = 1,
+        text = "Attend meetings",
+      ),
+    )
+
+    val someAssociationData = listOf(
+      uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.AdditionalConditionData(
+        id = 1,
+        field = "field1",
+        value = "value1",
+        sequence = 1,
+      ),
+      uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.AdditionalConditionData(
+        id = 2,
+        field = "numberOfCurfews",
+        value = "value2",
+        sequence = 2,
+      ),
+    )
+
+    val someAdditionalConditions = listOf(
+      uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.AdditionalCondition(
+        id = 1,
+        code = "associateWith",
+        sequence = 1,
+        text = "Do not associate with [NAME] for a period of [TIME PERIOD]",
+        expandedText = "Do not associate with value1 for a period of value2",
+        data = someAssociationData,
+        readyToSubmit = true,
+      ),
+    )
+
+    val someBespokeConditions = listOf(
+      uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.BespokeCondition(
+        id = 1,
+        sequence = 1,
+        text = "Bespoke one text",
+      ),
+      uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.BespokeCondition(
+        id = 2,
+        sequence = 2,
+        text = "Bespoke two text",
+      ),
+    )
+
+    val modelLicence = CrdLicence(
```

Comment:
Rather than building up this licence object could we use the crd licence that can be created from `TestData`?

Reply:
> Rather than building up this licence object could we use the crd licence that can be created from `TestData`?

No, the TestData returns CRDLicence type but we need the model licence type.